titan-server/src/accounts/file_entries.rs

```diff
@@ -48,26 +33,53 @@ pub fn find_most_recent(
 /// Creates a new file entry.
 pub fn create_file_entry(
     new_file_entry: &models::NewUserFileEntry,
-    titan_db: &MysqlConnection
-) -> Result<models::UserFileEntryWithType, diesel::result::Error> {
+    titan_db: &MysqlConnection,
+    wcf_db: &MysqlConnection,
+    app_config: &State<config::AppConfig>
+) -> Result<models::UserFileEntryAssoc, diesel::result::Error> {
     diesel::insert_into(schema::user_file_entries::table)
         .values(new_file_entry)
         .execute(titan_db)?;
 
     let last_inserted = find_most_recent(titan_db)?;
-    let entry_type = file_entry_types::find_by_id(
-        last_inserted.user_file_entry_type_id,
-        titan_db
-    )?;
 
-    Ok(models::UserFileEntryWithType {
-        id: last_inserted.id,
-        file_entry_type: entry_type,
-        user_id: last_inserted.user_id,
-        start_date: last_inserted.start_date,
-        end_date: last_inserted.end_date,
-        comments: last_inserted.comments,
-        date_modified: last_inserted.date_modified,
-        modified_by: last_inserted.modified_by
+    map_file_entry_assoc(last_inserted, titan_db, wcf_db, app_config)
+}
+
+pub fn map_file_entries_assoc(
+    file_entries: Vec<models::UserFileEntry>,
+    titan_db: &MysqlConnection,
+    wcf_db: &MysqlConnection,
+    app_config: &State<config::AppConfig>
+) -> Result<Vec<models::UserFileEntryAssoc>, diesel::result::Error> {
+    file_entries.into_iter().map(|entry| {
+        map_file_entry_assoc(entry, titan_db, wcf_db, app_config)
+    }).collect()
+}
+
+pub fn map_file_entry_assoc(
+    file_entry: models::UserFileEntry,
+    titan_db: &MysqlConnection,
+    wcf_db: &MysqlConnection,
+    app_config: &State<config::AppConfig>
+) -> Result<models::UserFileEntryAssoc, diesel::result::Error> {
+    let user = accounts::users::find_by_id(file_entry.user_id, titan_db)?;
+    let user_profile = accounts::users::map_user_to_profile(
+        user, wcf_db, app_config)?;
+    let file_entry_type = accounts::file_entry_types::find_by_id(
+        file_entry.user_file_entry_type_id, titan_db)?;
+    let modified_by_user = accounts::users::find_by_id(file_entry.modified_by, titan_db)?;
+    let modified_by_profile = accounts::users::map_user_to_profile(
+        modified_by_user, wcf_db, app_config)?;
```

Comment:
Any of the above failing results in no file entries being displayed. We should probably have a more graceful fallback to an unknown user or something of the like.

Reply:
I understand your reasoning behind implementing a fallback, but I see that as a band aid over an underlying problem that will likely resurface again in another form. These associations must *always* be valid. Cases where they are not hint to corrupt records that should either be corrected or discarded.